synapse/handlers/directory.py

```diff
@@ -307,15 +305,21 @@ def _update_canonical_alias(self, requester, user_id, room_id, room_alias):
             send_update = True
             content.pop("alias", "")
 
-        # Filter alt_aliases for the removed alias.
-        alt_aliases = content.pop("alt_aliases", None)
```

Comment:
My concern with this old code was that it would shadow a garbage `alt_aliases` value (of `None`), where in that case we don't want to modify the value.